spec/requests/registrations/user_profile_spec.rb

```diff
@@ -23,39 +23,104 @@
   end
 
   describe "POST /registrations/user-profile" do
-    it "creates an induction coordinator profile" do
-      post "/registrations/user-profile", params: { user: {
-        full_name: full_name,
-        email: email,
-      } }
-
-      expect(created_user.full_name).to eq(full_name)
-      expect(response).to redirect_to(:registrations_verification_sent)
-      expect(created_user.induction_coordinator_profile.schools).to include(school)
+    before do
+      mail = instance_double(ActionMailer::MessageDelivery, deliver_now: true)
+      allow(UserMailer).to receive(:confirmation_instructions).and_return(mail)
+    end
+
+    context "when request is valid" do
+      let(:send_request) do
+        post "/registrations/user-profile", params: { user: {
+          full_name: full_name,
+          email: email,
+        } }
+      end
+
+      it "creates an induction coordinator profile" do
+        send_request
+        expect(created_user.full_name).to eq(full_name)
+        expect(created_user.induction_coordinator_profile.schools).to include(school)
+      end
+
+      it "redirects to verification sent page" do
+        send_request
+        expect(response).to redirect_to(:registrations_verification_sent)
+      end
+
+      it "sends a confirmation email" do
+        send_request
+        expect(UserMailer).to have_received(:confirmation_instructions).with(created_user, /.*/, {})
```

Comment:
What are the params to `with` here?

Reply:
I should probably name them. They are a user, a token, and options.